libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridSmoothness.cpp

```diff
@@ -0,0 +1,93 @@
+//---- GPL ---------------------------------------------------------------------
+//
+// Copyright (C)  Stichting Deltares, 2011-2023.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// contact: delft3d.support@deltares.nl
+// Stichting Deltares
+// P.O. Box 177
+// 2600 MH Delft, The Netherlands
+//
+// All indications and logos of, and references to, "Delft3D" and "Deltares"
+// are registered trademarks of Stichting Deltares, and remain the property of
+// Stichting Deltares. All rights reserved.
+//
+//------------------------------------------------------------------------------
+
+#include <cmath>
+
+#include "MeshKernel/Constants.hpp"
+#include "MeshKernel/Vector.hpp"
+
+#include "MeshKernel/CurvilinearGrid/CurvilinearGridSmoothness.hpp"
+
+void meshkernel::CurvilinearGridSmoothness::Compute(const CurvilinearGrid& grid, const int direction, lin_alg::Matrix<double>& smoothness)
+{
+    smoothness.resize(grid.m_numM, grid.m_numN);
```

Comment:
You can use the utility function
```
    /// @brief         Resizes and fills a matrix
    /// @tparam        T        The data type of the matrix elements
    /// @tparam        storage  Matrix storage option
    /// @param[in,out] matrix   The matrix
    /// @param[in]     rows     The number of rows
    /// @param[in]     cols     The number of columns
    /// @param[in]     preserve Optional parameter that determines if the existing elements in
    ///                         the matrix should be preserved or overwritten by the provided
    ///                         or default (if not provided) fill value
    /// @param[in]     fill_value The optional fill value (defaults to the template class constructor)
    /// @return        True if the input matrix has been resized, false otherwise
    template <typename T, int storage>
    inline static bool ResizeAndFillMatrix(Matrix<T, storage>& matrix,
                                           Eigen::Index rows,
                                           Eigen::Index cols,
                                           bool preserve = false,
                                           T const& fill_value = {})
```

Reply:
done